src/batch/src/task/task_execution.rs

```diff
@@ -671,7 +672,7 @@ impl<C: BatchTaskContext> BatchTaskExecution<C> {
 
         let error = error.map(Arc::new);
         *self.failure.lock() = error.clone();
-        let err_str = error.as_ref().map(|e| e.to_string());
+        let err_str = error.as_ref().map(|e| e.to_report_string());
```

Comment:
If we really want to eagerly format the error, use `to_report_string` to replace `to_string`. This is basically an alias of `e.as_report().to_string()`.

Note that `to_string` implies `Display` trait. The behavior of `{}` `{:#}` `{:?}` `{:#?}` differ, and there're also corresponding `to_report_string` variants. Check the documentation of `Report`.